rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +303,151 @@ def query_rules_and_serialize(
+        counterparty = getattr(related_event, 'counterparty', None)
+        if counterparty == CPT_GAS:  # avoid checking the case of gas in evm events
+            continue
+
+        # query the accounting rule for the related event
+        cursor.execute(
+            'SELECT accounting_treatment FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))',  # noqa: E501
+            (
+                related_event.event_type.serialize(),
+                related_event.event_subtype.serialize(),
+                NO_ACCOUNTING_COUNTERPARTY if counterparty is None else counterparty,
+                NO_ACCOUNTING_COUNTERPARTY,
+            ),
+        )
+
+        if (raw_treatment := cursor.fetchone()) is not None and raw_treatment[0] is not None:
+            accounting_treatment = TxAccountingTreatment.deserialize_from_db(raw_treatment[0])
+            match accounting_treatment:
+                case TxAccountingTreatment.SWAP:
+                    # ensure that the event that we are going to check can exist
+                    # in the related events. Avoid index error
+                    if len(related_events) <= related_event_idx + 1:
+                        continue
+
+                    if (
+                        related_events[related_event_idx + 1].identifier == event.identifier
+                    ):
+                        return True
+                case TxAccountingTreatment.SWAP_WITH_FEE:
+                    if len(related_events) <= related_event_idx + 2:
+                        continue
+
+                    if event.identifier in {  # check both for the fee and the swap event
+                        related_events[related_event_idx + 1].identifier,
+                        related_events[related_event_idx + 2].identifier,
+                    }:
+                        return True
+
+        if isinstance(related_event, EvmEvent) is False:
+            continue  # only evm events have callbacks
+
+        # if there is no accounting treatment check for callbacks
+        if (callback_data := callbacks.get(related_event.get_type_identifier())) is None:
+            continue
+
+        affected_events_num, callback = callback_data
+        if affected_events_num == 0:
+            continue
+
+        # count the number of events that will be consumed
+        consumed_events = callback(
+            pot=pot,
+            event=related_event,  # type: ignore[arg-type] # mypy doesn't recognize that this is a evm event
+            other_events=iter(related_events[related_event_idx + 1:]),  # type: ignore[arg-type]  # mypy doesn't recognize that this is a evm event
+        )
+
+        # check if among the consumed events is the one that we are checking
+        return any(
+            event.identifier == consumed_event.identifier
+            for consumed_event in related_events[related_event_idx: related_event_idx + consumed_events + 1]  # noqa: E501
```

Comment:
```suggestion
            for consumed_event in related_events[related_event_idx: related_event_idx + processed_events_num]  # noqa: E501
```

Reply:
1. you won't need this with the changes I recommended.
2.  I removed the +1 since you will have +1 on the processed_events_num